NumPy.py

```diff
@@ -1,12 +1,236 @@
+    :return: The matrix that created.
+    """
+    matrix = np.ones((10, 10))
+    matrix[1:-1, 1:-1] = 0
+    return matrix
+
+
+# ----------q5----------
+def question5(matrix, vector):
+    """
+    The function adds the vector to each row of a given matrix.
+    :param matrix: The given matrix.
+    :param vector: The vector that adds.
+    :return: The matrix after change.
+    """
+    matrix_after_add = np.empty_like(matrix)
+    for row in range(len(matrix)):
+        matrix_after_add[row, :] = matrix[row, :] + vector
+
+    return matrix_after_add
+
+
+# ----------q6----------
+def question6(x, y):
+    """
+    The function computes the x and y coordinates for points on a sine curve and plot the points using matplotlib.
+    :param x: coordinate x.
+    :param y: coordinate y.
+    """
+    x = np.arange(x, y * np.pi, 0.1)
+    y = np.sin(x)
+    plt.plot(x, y)
+    plt.show()
+
+
+# ----------q7----------
+def question7():
+    """
+    The function creates a 4x4 array with random values, after creates a new array from the said array swapping first
+    and last rows.
+    :return: The matrix after change.
+    """
+    matrix = np.random.rand(4, 4)
+    return matrix[::-1]
+
+
+# ----------q8----------
+def question8(matrix, number):
+    """
+    The function replaces all numbers in a given array which is equal, less and greater to a given number.
+    The function prints the results.
+    :param matrix: The given matrix.
+    :param number: The number with the exchange is made.
+    """
+    number_to_replace = 0
+
+    print(np.where(matrix == number, number_to_replace, matrix))
+    print(np.where(matrix < number, number_to_replace, matrix))
+    print(np.where(matrix > number, number_to_replace, matrix))
+
+
+# ----------q9----------
+def question9(matrix1, matrix2):
+    """
+    The function multiplies two given arrays of same size element-by-element.
+    :param matrix1: The first given matrix.
+    :param matrix2: The second given matrix.
+    :return: The multiplied matrix.
+    """
+    return np.multiply(matrix1, matrix2)
+
+
+# ----------q10----------
+def question10(matrix):
+    """
+    The function sorts an along the first, last axis of an array.
+    :param matrix: Te given matrix.
+    """
+    first_axis_sorted = np.sort(matrix, axis=0)
+    print(first_axis_sorted)
+    last_axis_sorted = np.sort(first_axis_sorted, axis=1)
+    print(last_axis_sorted)
+
+
+# ----------q11----------
+def question11():
+    """
+    The function creates a 3-D array with ones on a diagonal and zeros elsewhere.
+    :return: The 3-D array.
+    """
+    return np.eye(3)
+
+
+# ----------q12----------
+def question12(matrix):
+    """
+    The function removes single-dimensional entries from a specified shape.
+    :param matrix: The given matrix.
+    :return: The matrix after change.
+    """
+    return np.squeeze(matrix)
+
+
+# ----------q13----------
+def question13(*arrays):
```

Comment:
Here the imprtance of type hinting is clear, your function can be overloaded with infinte numpy arrays.